### ISSUE TITLE
`indent-binary-ops`: treats TS union types as binary operators

### ISSUE THREAD

**Issue by danyadev** [eslint-stylistic/eslint-stylistic#676]
### Validations

- [x] Follow our [Code of Conduct](https://github.com/eslint-stylistic/eslint-stylistic/blob/main/CODE_OF_CONDUCT.md)
- [x] Read the [Contributing Guide](https://eslint.style/contribute/guide).
- [x] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.

### Describe the bug

What my code looks like and what the rule wants from it:  
<img width="415" alt="Image" src="https://github.com/user-attachments/assets/951f592d-7bb7-4203-ac1b-13b4d5a7a0b4" />

Union types shouldn't be treated as binary operations and the rule shouldn't touch them

$\textsf{\color{gray}(btw the indent rule also has problems with this style of unions which cannot be suppressed by igronedNodes)}$

### Reproduction

https://stackblitz.com/edit/github-y4yk9nxw?file=eslint.config.js,src%2Findex.ts

### Contributes

- [ ] I am willing to submit a PR to fix this issue
- [ ] I am willing to submit a PR with failing tests

**Comment by Jannchie** [comment]
I’m not sure if this is related, but I ran into an issue: if I construct a nested ternary conditional type combined with a binary operator (Intersection/Union Types?) like the following, it will trigger circular fixes between `style/indent` and `style/indent-binary-ops`.

```ts
export type Options =
  unknown extends unknown
    ? unknown extends unknown
      ? unknown
      & unknown
      : unknown
    : unknown;
```

**Comment by DamianGlowala** [comment]
I second @Jannchie - having the exact same issue where two different rules mentioned above contradict each other.

cc @9romise

Another example where these fail (notice `// eslint-disable-next-line @stylistic/indent`):
```
type TVReturnType<
    TSection extends string,
    V extends TVVariants<TSection, S>,
    S extends TVSlots<TSection>,
    B extends ClassValue,
    EV extends TVVariants<TSection, ES>,
    ES extends TVSlots<TSection>,
    // @ts-expect-error external types
    E extends TVReturnType = undefined,
> = {
    (props?: TVProps<TSection, V, S, EV, ES>): HasSlots<TSection, S, ES> extends true
    // eslint-disable-next-line @stylistic/indent
    ? {
            // eslint-disable-next-line @typescript-eslint/no-empty-object-type
            [K in keyof (ES extends undefined ? {} : ES)]: (
                slotProps?: TVProps<TSection, V, S, EV, ES>,
            ) => string;
        } & {
            // eslint-disable-next-line @typescript-eslint/no-empty-object-type
            [K in keyof (S extends undefined ? {} : S)]: (slotProps?: TVProps<TSection, V, S, EV, ES>) => string;
        } & {
            // @ts-expect-error external types
            // eslint-disable-next-line @typescript-eslint/no-empty-object-type
            [K in TVSlotsWithBase<TSection, {}, B>]: (slotProps?: TVProps<TSection, V, S, EV, ES>) => string;
        }
        : string
} & TVReturnProps<TSection, V, S, B, EV, ES, E>
```

**Comment by brainkim** [comment]
Same rule removes correct indentation. `@stylistic/eslint-plugin@5.10.0`, `indent-binary-ops` as the only rule enabled:

```js
rules: {"@stylistic/indent-binary-ops": ["error", "tab"]}
```

```ts
export const Portal = Symbol.for("crank.Portal") as unknown as Component<{
	root?: object;
}> &
	symbol;
```

```
6:1  error  Expected indentation of 0 tabs
```

`--fix` gives:

```ts
export const Portal = Symbol.for("crank.Portal") as unknown as Component<{
	root?: object;
}> &
symbol;
```

Only happens when the left operand is multiline — the rule anchors to `}> &` at column 0.
